src/conways-game-of-life/index.ts

```diff
@@ -0,0 +1,71 @@
+const updateCellStatus = (currentCellStatus: number, neighbourSum: number) => {
+    if (currentCellStatus === 1 && neighbourSum > 3) {
+        return 0;
+    }
+    else if (currentCellStatus === 1 && neighbourSum < 2) {
+        return 0;
+    }
+    else if (currentCellStatus === 0 && neighbourSum === 3) {
+        return 1;
+    }
+    return currentCellStatus;
+} //Apply the conditions for updating the status of a cell 
+
+const calculateNeighbourSum = (grid: number[][], colIndex: number, rowIndex: number) => {
+    let neighbourSum = 0;
+    for (let i = colIndex - 1; i <= colIndex + 1; i++) {
+        for (let j = rowIndex - 1; j <= rowIndex + 1; j++) {
+            if (i === colIndex && j === rowIndex) continue;
+            if (i >= 0 && i < grid[0].length && j >= 0 && j < grid.length) {
+                neighbourSum += grid[j][i];
+            }
+        }
+    }
+    return neighbourSum;
+} // Calculates and return the sum of neighbors alive around the cell of interest
```

Comment:
I can see some improvements. However, I think that there are room for improvement.

Based on the code you are looping through the whole grid and checking only the values that are within the bound you want, if so you increment the number of neighbours. Again, is a 5x5 grid it's not a real issue but in a 100x100 grid it can grow fast. Imagine in a 1,000,000 x 1,000,000 grid?

I'd suggest you only loop through the values you need. For example, instead of `let i = colIndex - 1` check if  `colIndex - 1` is within bounds, if not set it to the minimum value that you have in bounds. The same applies to `i <= colIndex + 1` if the `colIndex + 1` is out of bound, set it the max value in bound. Once you do that, repeat it for `j`.

Check: 
1. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/min
2. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max